schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -1,549 +1,848 @@
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "allOf": [
+              {
+                "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/resourceBase"
+              },
+              {
+                "oneOf": [
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2014-10-01-preview/Microsoft.Authorization.json#/definitions/roleAssignments"
+                  },
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Authorization.json#/definitions/locks"
+                  },
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Resources.json#/definitions/links"
+                  },
+                  {
+                    "$ref": "#/resourceDefinitions/extensionsChild"
+                  }
+                ]
+              }
             ]
+          },
+          "description": "Microsoft.Compute/virtualMachines: Resource Definition for Virtual Machines."
         }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "properties"
+      ]
     },
-    "definitions": {
-        "id": {
-            "properties": {
-                "id": {
-                    "type": "string"
-                }
-            },
-            "type": "object",
-            "required": [
-                "id"
-            ]
+    "virtualMachineScaleSets": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.Compute/virtualMachineScaleSets"
+          ]
         },
-        "networkInterfaces": {
-            "properties": {
-                "id": {
-                    "type": "string"
-                },
-                "properties": {
-                    "properties": {
-                        "primary": {
-                            "type": "boolean"
-                        }
-                    },
-                    "type": "object",
-                    "required": [
-                        "primary"
-                    ]
-                }
+        "apiVersion": {
+          "enum": [
+            "2015-05-01-preview",
+            "2015-06-15"
+          ]
+        },
+        "properties": {
+          "sku": {
+            "$ref": "#/definitions/sku"
+          },
+          "properties": {
+            "upgradePolicy": {
+              "$ref": "#/definitions/upgradePolicy"
             },
-            "type": "object",
-            "required": [
-                "id"
+            "virtualMachineProfile": {
+              "$ref": "#/definitions/virtualMachineProfile"
+            }
+          },
+          "type": "object",
+          "required": [
+            "upgradePolicy",
+            "virtualMachineProfile"
+          ]
+        },
+        "resources": {
```

Comment:
If virtualMachineScaleSets does not have any child resources other than the default ones, you do not need to add "resources" here. Please remove the "resources" property.